rfcs/0128-selective-auto-merge-on-bot-upgrades.md

```diff
@@ -0,0 +1,58 @@
+---
+feature: Selective auto-merge on bot upgrades
+start-date: 2022-07-07
+author: superherointj
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a `meta.autoMerge` attribute to packages to allow committers to delegate merge rights to package maintainers for bot auto updates.
+
+# Motivation
+[motivation]: #motivation
+
+* Reduce pending PRs for review and merge in nixpkgs.
+* Save reviewers and commiters time.
+* Speed up package upgrades.
+* Motivate reviewers into becoming maintainers and reviewing PRs.
+
+# Detailed design
+[design]: #detailed-design
+
+Add a new `meta.autoMerge` package attribute with type `bool` defaulting to `false`. To be documented in Nixpkgs manual.
```

Comment:
If such a attribute/flag is to be implemented, I don't think it should be a `meta` ket, but a `passthru` one.

`meta`, as the name suggests, it for metadata about the package, not about how our particular infrastructure would treat such package.

Even if some `meta` flags are useful to our infra - `license` defines if something can be cached or not, but has an informational nature independent of it; the same can be said about `broken` and `badPlatforms`.

Maybe `meta.hydraXXX` is a notable exception.